### ISSUE TITLE
Allow dataset to not load a template

### ISSUE THREAD

**Issue by ruflin** [elastic/package-spec#26]
There are some datasets which allow to select and configure and input directly on the Agent side and with it, configure the data_stream.dataset field. Because of this, loading a template does not make sense in this context.

Our first example of this is the `log` package. It allows to tail any log file and the user can configure where the log should be shipped to. It currently loads a template into `logs-log.log-*`, but the template that actually applies to the data is the `logs-*-*` template except by chance `log.log` was set as dataset name.

We currently validate that dataset MUST contain the `data_stream.*` fields. But in this case the creator of the package should be able to set a flag in the `daaset/manifest.yml` file like `template: false` (better name suggestions welcome) which disables this validation. And then on the Kibana side, if no fields.yml exists, no template is created.

**Comment by ruflin** [comment]
@ycombinator I assume this would also be a change to the package-spec?

**Comment by ycombinator** [comment]
> We currently validate that dataset MUST contain the `data_stream.*` fields. 

Quick clarification. I don't see any `data_stream` fields in the current spec for the dataset manifest file: https://github.com/elastic/package-spec/blob/master/versions/1/dataset/manifest.spec.yml. But perhaps by "current" you mean "soon", as in the change we're making here: https://github.com/elastic/package-spec/pull/24/files#diff-13d2900dcce8e312a06c834b665816a1R71?

**Comment by ruflin** [comment]
Yes, but I need my guess is that in package-spec we don't even enforce this today and only the package-registry does for historical reasons. I would have to check the code to say for sure.

**Comment by ycombinator** [comment]
In the change we're making here: https://github.com/elastic/package-spec/pull/24/files#diff-13d2900dcce8e312a06c834b665816a1R71, the `data_stream` field is of type `string`. So what did you mean by "dataset MUST contain the `data_stream.*` fields". Specifically, why the `*`, since it's not an object field? Just want to make sure we're thinking of the same `data_stream` field (especially now that everything is called data streams 🙂).

**Comment by ruflin** [comment]
@ycombinator I think we talk about 2 different things. I'm referring to the check that the base_fields must exist https://github.com/elastic/package-storage/blob/production/packages/apache/0.1.4/dataset/access/fields/base-fields.yml which are used for creating a template. It also means, a template is always created but this should be optional.

**Comment by ycombinator** [comment]
Ah, yes, that's different 🙂. Currently the spec expects that each dataset folder must contain a `fields` sub-folder:

https://github.com/elastic/package-spec/blob/d669841b7963c3779e410c9ce581cfb48a66ee97/versions/1/dataset/spec.yml#L16-L20

And, further, it expects that this `fields` sub-folder must contain at least one YAML file:

https://github.com/elastic/package-spec/blob/d669841b7963c3779e410c9ce581cfb48a66ee97/versions/1/dataset/fields/spec.yml#L4-L9

It does **not** specifically require that this file be named `base-fields.yml`. 

In this context, what is the change you are proposing to the spec?

**Comment by ruflin** [comment]
There should be an option, to create a dataset (in the future data_stream) that does not contain the `fields` directory and with it not template is loaded. I would expect that this is specified in the dataset with a config option something like `template: false` or `fields: false`. With this, we can still validate all the other casese.

**Comment by ycombinator** [comment]
Thanks for clarifying. I understand now.  Personally I slightly prefer `fields: false` since the folder name corresponding to it is also `fields`.

Concretely I think this means the following enhancements to the spec:
1. Allow for an optional `fields` boolean property (default value, `true`) under https://github.com/elastic/package-spec/blob/d669841b7963c3779e410c9ce581cfb48a66ee97/versions/1/dataset/manifest.spec.yml#L70,
2. Make the `fields` folder optional here: https://github.com/elastic/package-spec/blob/d669841b7963c3779e410c9ce581cfb48a66ee97/versions/1/dataset/spec.yml#L19, and
3. Add a [semantic validation](https://github.com/elastic/package-spec/issues/47) rule that first checks the value of the `fields` boolean property (from 1. above). If it is absent or set to `true`, check that the `fields` folder is present (essentially overriding 2. above).

@ruflin @mtojek does this make sense to you? If yes, I'll add this issue to the list in https://github.com/elastic/package-spec/issues/47. If not, I'm open to other ways of going about this change + it's enforcement in the spec.

**Comment by ruflin** [comment]
@ycombinator spot on.
